doc/setup.rst

```diff
@@ -198,6 +198,55 @@ To help expose these tracepoints (and any additional one we might require in
 the future) as trace events, an external module is required and is provided
 under the name of "lisa" in $LISA_HOME/tools/kmodules/lisa
 
+Pre-requisites
```

Comment:
This section should go under "Manual route". We really don't want people to start getting into that unless they really have to, so the first thing the user must read is:
> LISA Python package will compile and load the module automatically when required for tracing so there is usually no reason to do so manually

This should be properly "under" that manual section as this is a child of it, it is not a sibling.

Reply:
The pre-requisites are not related to the build method, they apply to the automatic build method just as much as they do to the manual.

Putting the pre-requisites for the automatic build under the manual build section makes no sense, since it applies to both it should be a sibling section that comes before the two imo.

Without following the steps in pre-requisites the automatic method will fail on every Android build - because of the symbols and CFI and product kernels and because of CFI on mainline ones.

The automatic method without pre-requisites pretty much only works on Juno and some servers that can run pure mainline Arm kernels, very few devices in the grand scheme of things.

Reply:
Ok that makes sense.

>pretty much only works on Juno and some servers that can run pure mainline Arm kernels, very few devices in the grand scheme of things.

There are hundreds if not thousands of devices that fall into that category. Any amd64 desktop, laptop, server in the past 17 years basically. We just happen to be more interested in the few locked down devices android phones has brought in the recent years, but that's hardly the majority of devices, and that's definitely not the focus of the entire LISA userbase so we very much want to keep android-isms as a layer on top of the rest, not as a base assumption.